main.py

```diff
@@ -1,16 +1,37 @@
-# This is a sample Python script.
+class Product:
+    def __init__(self, name: str, price: int) -> None:
+        self.name = name
+        self.price = price
 
-# Press Shift+F10 to execute it or replace it with your code.
-# Press Double Shift to search everywhere for classes, files, tool windows, actions, and settings.
+    def product_show(self):
+        return f"{self.name} price is {self.price}$"
 
+    def all_price(self, quantity: int):
+        return round((self.price * quantity), 2)
 
-def print_hi(name):
-    # Use a breakpoint in the code line below to debug your script.
-    print(name )    # Press Ctrl+F8 to toggle the breakpoint.
+class ShoppingCart:
+    def __init__(self) -> None:
+        self.products = []
+        self.quantity = []
 
+    def add_product(self, product: Product, quantity):
+        self.products.append(product)
+        self.quantity.append(quantity)
 
-# Press the green button in the gutter to run the script.
-if __name__ == '__main__':
-    print_hi('PyCharm')
+    def get_total(self):
+        zipper = list(zip(self.products, self.quantity))
+        total = 0
+        for i in range(len(zipper)):
```

Comment:
`zip` сам по собі є генератором. Не треба додатково перетворювати його на список.
Також, будь який генератор одночасно є ітератором (але не навпаки).
Отже можна ітерувати одразу за змістом генератора.
```suggestion
        for product, quantity in zip(self.products, self.quantity):
```